test/e2e/lb_subnets_test.go

```diff
@@ -0,0 +1,419 @@
+		}
+	}); err != nil {
+		t.Fatalf("failed to update ingresscontroller: %v", err)
+	}
+
+	// Verify the subnet annotation is removed on the service.
+	if err := waitForLBSubnetAnnotation(t, ic, nil, lbService.Status.LoadBalancer); err != nil {
+		t.Fatalf("failed to wait for expected subnet annotation on service: %v", err)
+	}
+
+	// Expect the load balancer to provision successfully with the subnets removed.
+	if err := waitForIngressControllerCondition(t, kclient, 10*time.Minute, icName, availableConditionsForIngressControllerWithLoadBalancer...); err != nil {
+		t.Fatalf("failed to observe expected conditions: %v", err)
+	}
+
+	// Verify the subnets status field is configured to what we expect.
+	if err := verifyIngressControllerSubnetStatus(t, icName); err != nil {
+		t.Fatalf("failed to verify ingresscontroller subnet status: %v", err)
+	}
+}
+
+// waitForLBSubnetAnnotation waits for the provided subnets to appear on the LoadBalancer-type service for the
+// IngressController. It will return an error if it fails to observe the updated subnets.
+func waitForLBSubnetAnnotation(t *testing.T, ic *operatorv1.IngressController, subnets *operatorv1.AWSSubnets, oldLoadBalancerStatus v12.LoadBalancerStatus) error {
+	t.Helper()
+	lbService := &v12.Service{}
+	expectedSubnetAnnotation := ingress.JoinAWSSubnets(subnets, ",")
+	t.Logf("waiting for %q service to have subnet annotation of %q", controller.LoadBalancerServiceName(ic), expectedSubnetAnnotation)
+	err := wait.PollUntilContextTimeout(context.Background(), 10*time.Second, 5*time.Minute, false, func(ctx context.Context) (bool, error) {
+		if err := kclient.Get(ctx, controller.LoadBalancerServiceName(ic), lbService); err != nil {
+			t.Logf("get %q failed: %v, retrying ...", controller.LoadBalancerServiceName(ic), err)
+			return false, nil
+		}
+		if reflect.DeepEqual(lbService.Status.LoadBalancer, oldLoadBalancerStatus) {
+			t.Logf("waiting for service %q to be updated", controller.LoadBalancerServiceName(ic))
+			return false, nil
+		}
+		val, ok := lbService.Annotations[awsLBSubnetAnnotation]
+		// Handle the case where subnets are expected to be nil (annotation should not exist).
+		if subnets == nil {
+			if ok {
+				return true, fmt.Errorf("expected %q annotation to be removed got %q", awsLBSubnetAnnotation, val)
+			}
+			return true, nil
+		}
+		// Handle the case where subnets are expected (annotation should exist and match).
+		if !ok || val != expectedSubnetAnnotation {
+			return true, fmt.Errorf("expected %q annotation %q got %q", awsLBSubnetAnnotation, expectedSubnetAnnotation, val)
+		}
+		return true, nil
+	})
+	if err != nil {
+		return fmt.Errorf("error updating the %q service: %v", controller.LoadBalancerServiceName(ic), err)
+	}
+	return nil
+}
+
+// verifyIngressControllerSubnetStatus verifies that the IngressController subnet status fields are
+// equal to the appropriate subnet spec fields when an IngressController is in a Progressing=False state.
```

Comment:
We aren't, I listed this just as a function "pre-condition", i.e. this is a restriction or scenario to use this function.

But it's a good point, we can enforce Progressing=False just to be absolutely sure. The function would fail anyways, but might be helpful to fail quicker. I'll add.